zipline/algorithm.py

```diff
@@ -485,16 +506,15 @@ def run(self, data=None, overwrite_sim_params=True):
                 copy_panel.items = self._write_and_map_id_index_to_sids(
                     copy_panel.items, copy_panel.major_axis[0],
                 )
-                assets = self.trading_environment.asset_finder.retrieve_all(
-                    copy_panel.items)
+                self._assets_from_source = \
+                    self.trading_environment.asset_finder.retrieve_all(
+                        copy_panel.items
+                    )
                 equities = []
-                futures = []
-                for asset in assets:
+                for asset in self._assets_from_source:
                     if isinstance(asset, Equity):
                         equities.append(asset)
-                    elif isinstance(asset, Future):
-                        futures.append(asset)
-                if equities:
+                if self._equities:
```

Comment:
Where is this defined?  Do we no longer have any tests that run with a DataFrame?